test/e2e/tests/conftest.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+import docker
+import os
+from time import sleep
+from sinspqa import LOGS_PATH
+from sinspqa.docker import get_container_id
+
+
+@pytest.fixture(scope="session", autouse=True)
+def docker_client():
+    """
+    Create a docker client to be used by the tests.
+
+    Returns:
+        A docker.DockerClient object created from the environment the tests run on.
+    """
+    return docker.from_env()
+
+
+@pytest.fixture(scope="module")
+def tester_id(docker_client: docker.client.DockerClient):
+    """
+    Get the truncated ID of the test runner.
+
+    Returns:
+        A 12 character string with the ID.
+    """
+    tester_container = docker_client.containers.get('sinsp-e2e-tester')
+
+    return get_container_id(tester_container)
+
+
+def wait_container_running(container: docker.models.containers.Container, additional_wait: int = 0, retries: int = 5):
+    success = False
+
+    for _ in range(retries):
+        container.reload()
+
+        if container.status == 'running':
+            success = True
+            break
+
+        sleep(0.5)
+
+    if not success:
+        raise TimeoutError
+
+    if additional_wait:
+        sleep(additional_wait)
+
+
+@pytest.fixture(scope="function")
+def run_containers(request, docker_client: docker.client.DockerClient):
+    """
+    Runs containers, dumps their logs and cleans'em up
+    """
+    containers = {}
+    post = {}
+
+    for name, container in request.param.items():
+        image = container['image']
+        args = container.get('args', '')
+        privileged = container.get('privileged', False)
+        mounts = container.get('mounts', [])
+        environment = container.get('env', {})
+        additional_wait = container.get('init_wait', 0)
+        post_validation = container.get('post_validation', None)
+        stop_signal = container.get('signal', None)
+
+        handle = docker_client.containers.run(
+            image,
+            args,
+            name=name,
+            detach=True,
+            privileged=privileged,
+            mounts=mounts,
+            environment=environment
+        )
+
+        containers[name] = handle
+        post[name] = {
+            'validation': post_validation,
+            'signal': stop_signal
+        }
+
+        wait_container_running(handle, additional_wait)
+
+    yield containers
+
```

Comment:
It looks like the cleanup code below can't run if an exception is raised by `wait_container_running` 😢, which would invalidate all subsequent tests (see PR comment)